### PR TITLE
Convert second chunk of integration tests to RSpec

### PR DESCRIPTION
**Pull request by hannako**
## What

Following on from  https://github.com/alphagov/collections/pull/2320

Convert the second batch of integration tests. 

Trello: [Card](https://trello.com/c/CimzmpjN/1427-convert-integrations-tests-to-rspec)
Issue: #2259 

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

### CODE REVIEW

File: spec/features/step_nav_page_spec.rb [hunk 1]
```diff
@@ -16,22 +16,22 @@ class StepNavPageTest < ActionDispatch::IntegrationTest
   end
 
   it "renders breadcrumbs" do
-    assert page.has_css?(".gem-c-breadcrumbs")
+    expect(page.has_css?(".gem-c-breadcrumbs")).to be
```
Comment (hannako) [inline spec/features/step_nav_page_spec.rb]: this is fixed in the convert commit

File: spec/features/world_wide_taxon_browsing_spec.rb [hunk 2]
```diff
@@ -65,36 +65,36 @@ def when_i_visit_that_taxon
...
 
   def then_i_see_the_taxon_page
-    assert page.has_selector?("title", text: "Japan", visible: false)
+    expect(page).to have_selector("title", text: "Japan", visible: false)
   end
 
   def and_i_can_see_the_content_tagged_to_the_taxon
     tagged_content.each do |content|
-      assert page.has_link?(content["title"])
+      expect(page.has_link?(content["title"])).to be
```
Comment (sihugh) [inline spec/features/world_wide_taxon_browsing_spec.rb:73]: .to be

File: spec/features/world_wide_taxon_browsing_spec.rb [hunk 3]
```diff
@@ -65,36 +65,36 @@ def when_i_visit_that_taxon
...
     end
   end
 
   def and_i_can_see_the_email_signup_link
-    assert page.has_link?("Get emails for this topic")
+    expect(page.has_link?("Get emails for this topic")).to be
```
Comment (sihugh) [inline spec/features/world_wide_taxon_browsing_spec.rb:78]: .to be

**Review by sihugh — CHANGES_REQUESTED** [review]
Just a few of the weird `.to be` conversions. Otherwise, all good

File: spec/features/transition_landing_page_spec.rb [hunk 4]
```diff
@@ -1,7 +1,6 @@
-require "integration_test_helper"
-require_relative "../support/transition_landing_page_steps"
+require "integration_spec_helper"
 
-class TransitionLandingPageTest < ActionDispatch::IntegrationTest
+feature "Transition landing page" do
```
Comment (sihugh) [inline spec/features/transition_landing_page_spec.rb:3]: RSpec.

**Review by sihugh — APPROVED** [review]
🎉

**Review by sihugh — APPROVED** [review]

**Pull request merged**